### PR TITLE
[WIP] Draft of events and logs in the security guide

### PR DESCRIPTION
**Pull request by rhcarvalho**
This adds a new topic in the existing Security Guide, focusing on node
and system events and logs from main platform components.

Trello card: https://trello.com/c/uI7tGkmM

### CODE REVIEW

**Comment by rhcarvalho** [comment]
@brenton @sosiouxme would like to validate the direction of this.

File: security/monitoring.adoc [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+Link back to logging stack
+
+Others: build logs, deployment logs, ...
+
+- How to discover malicious intent from logs(?)
+
+
+== What to look for
+
+// How to tie API calls back to cluster users? Via Token?
```
Comment (sosiouxme) [inline security/monitoring.adoc]: probably the audit log

File: security/monitoring.adoc [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+- Top API Server consumers
+- API requests by IP: high usage from a single IP may reveal a misbehaving user or misconfigured service
```
Comment (sosiouxme) [inline security/monitoring.adoc]: IP is likely to be a load balancer or proxy and not indicate much about original user. audit log by user is probably the direction to go.

File: security/monitoring.adoc [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+- Recent API requests with users having cluster-admin privileges
```
Comment (sosiouxme) [inline security/monitoring.adoc]: yes... actually, just retrieving a list of who has cluster-admin (or other high-level) privs is probably useful. and not as straightforward as you might think, though i guess for simplicity we can just check group membership e.g.:

```
I0907 17:24:13.550666       1 audit.go:133] 2017-09-07T17:24:13.550647939Z AUDIT: id="1a0ce9e6-b4cc-4afb-b10a-8769b3b21678" ip="172.18.6.149" method="DELETE" user="system:admin" groups="\"system:cluster-admins\"
I0907 17:24:13.553212       1 audit.go:48] 2017-09-07T17:24:13.553202441Z AUDIT: id="1a0ce9e6-b4cc-4afb-b10a-8769b3b21678" response="200"
```

(there are probably still other ways to grant privileges without putting the user in a group)

**Review by sosiouxme — COMMENTED** [review]
A few things to remark on. I'm not sure what the overall direction should be here; maybe "here are things to configure and monitor" and a few specific examples. It feels like something where we want to try to enable people with a real vested interest in security to discover what to look for and hopefully report back with their findings so we can incorporate it later.

File: security/monitoring.adoc [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+Use `*journalctl*`
+
+- API Server logs
+- Kubelet logs
+- Proxy logs
+
+Link back to logging stack
+
+Others: build logs, deployment logs, ...
```
[the remark this answers is not in the document]
Comment (sosiouxme) [reply security/monitoring.adoc]: OpenSCAP... guess we can leave it to CloudForms

**Comment by vikram-redhat** [comment]
[Document: [SCCFSI] declare which events are related to security](https://trello.com/c/FL5cfxcp/598-document-sccfsi-declare-which-events-are-related-to-security)

**Comment by tmorriso-rh** [comment]
Hi @rhcarvalho, I am following up on this PR. Is this the information you are proposing to add to our existing [Security Guide](https://docs.openshift.com/container-platform/3.6/security/index.html)?

Is this information still current?

If so, what OCP release(s) does this change apply to? Thanks.

**Comment by brenton** [comment]
Hi @tmorriso-rh, we're going to have to hand this off to someone else.  It's also going to need a bit more work.

**Comment by tmorriso-rh** [comment]
Thanks @brenton 

@vikram-redhat, please see the updated comment about the status of this PR. Thanks.

**Comment by tmorriso-rh** [comment]
@brenton Hi, do you know who will take ownership of this topic? It is assigned to our current sprint, which ends on Friday, October 6. We can move it to the backlog if the team needs more time. Thanks.

**Comment by brenton** [comment]
@tmorriso-rh Unfortunately, more time will definitely be needed.  I'm going to bring this situation up to PM.

**Comment by sosiouxme** [comment]
As @rhcarvalho is no longer on the team and I'm working on the equivalent PR, this can be closed.

**Pull request closed without merging**